yjit/src/codegen.rs

```diff
@@ -2450,45 +2450,33 @@ fn gen_setinstancevariable(
         None
     };
 
-    // Get the next shape information if it needs transition
+    // The current shape doesn't contain this iv, we need to transition to another shape.
     let new_shape = if !shape_too_complex && receiver_t_object && ivar_index.is_none() {
-        let shape = comptime_receiver.shape_of();
-
-        let current_capacity = unsafe { (*shape).capacity };
-
-        // If the object doesn't have the capacity to store the IV,
-        // then we'll need to allocate it.
-        let needs_extension = unsafe { (*shape).next_iv_index >= current_capacity };
+        let current_shape = comptime_receiver.shape_of();
+        let dest_shape = unsafe { rb_shape_get_next(current_shape, comptime_receiver, ivar_name) };
```

Comment:
Small detail, but would use the name `next_shape` since the function is called `rb_shape_get_next`